x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/container/use_container_metrics_table.ts

```diff
@@ -81,6 +87,25 @@ export function useContainerMetricsTable({
   };
 }
 
+function addEventModuleFilter(
+  filterClauseDsl: QueryDslQueryContainer | undefined
+): QueryDslQueryContainer {
+  return {
+    bool: {
+      filter: [
+        {
+          term: {
+            'event.module': 'kubernetes',
+          },
+        },
+        {
+          ...filterClauseDsl,
+        },
```

Comment:
When `typeof filterClauseDsl === 'undefined'`, then this would put an empty object into the `filter` array. How about we avoid adding it in the first place?

```suggestion
        ...(filterClauseDsl != null ? [filterClauseDsl] : []),
```

Reply:
good catch! I'll just your suggestion for an array with an object

Reply:
but we're merging into an array? :thinking:

Reply:
ah, I see now. At first glance I thought you were adding it to the filter array. You are actually concatenating. Got it